app/Http/Controllers/Auth/RegisterController.php

```diff
@@ -48,25 +49,44 @@ public function __construct()
      */
     protected function validator(array $data)
     {
+        info($data);
         return Validator::make($data, [
             'name' => ['required', 'string', 'max:255'],
-            'email' => ['required', 'string', 'email', 'max:255', 'unique:users'],
-            'password' => ['required', 'string', 'min:8', 'confirmed'],
+            'email' => ['required', 'string', 'email', 'max:255',], //'unique:users'],
+            'password' => ['required', 'string', 'min:8'],
         ]);
     }
 
     /**
      * Create a new user instance after a valid registration.
      *
      * @param  array  $data
-     * @return \App\User
+     * @return \App\Models\User
      */
-    protected function create(array $data)
+    protected function create(RegisterRequest $request)
+    {
+        $data = request(['name', 'email', 'password']);
+        $validatedData = $request->validated();
+        if ($validatedData){
+            $user = User::create([
+                'name' => $data['name'],
+                'email' => $data['email'],
+                'password' => Hash::make($data['password']),
+            ]);
+            //log in user after registration
+            $credentials = ['email'=>$request->email, 'password'=>$request->password];
+            $token = auth()->attempt($credentials);
+            return $this->respondWithToken($token);
+        }else{
+            return response()->json(['error' => 'Bad request'], 400);
+        }
+    }
```

Comment:
Ovdje vec imas ovaj validator koji se poziva preko Auth ruta, ako se dobro sjecam ukoliko je to slucaj onda ti je ovaj custom request `RegisterRequest ` samo nepotreban posao, jer vec bude izvaldiran dok dodje do $request->validated();